### PR TITLE
Add Submissions table to database and create controller

### PR DESCRIPTION
**Pull request by ErlingHauan**
* Added a Submissions table to the DbContext.
* Added new models: SubmissionEntity, SubmissionDto and SubmissionResponse.
* Added a SubmissionsController and a SubmissionRepository.
* Added endpoints for creating, reading, and deleting submissions.
* Added tests for the endpoints.
* Added seed data for the new table.

These endpoints were added:
image

### CODE REVIEW

File: backend/src/Repositories/SubmissionRepository.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    public async Task<SubmissionEntity> GetSingle(Guid submissionId)
+    {
+        var entity = await _context.Submissions.FindAsync(submissionId);
+
+        if (entity == null)
+        {
+            Console.WriteLine($"Submission with ID {submissionId} was not found.");
+            return null;
```
Comment (JamalAlabdullah) [inline backend/src/Repositories/SubmissionRepository.cs:39]: We have a warning here that the method maybe return null , and we do not allow for that, so we can add `nullable `type `?` to SubmissionEntity in line 32.

File: backend/src/Repositories/SubmissionRepository.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        _context = context;
+    }
+
+    public async Task<List<SubmissionEntity>> GetAll()
+    {
+        var submissionList = await _context.Submissions.ToListAsync();
+        return submissionList;
+    }
+
+    public async Task<SubmissionEntity> GetSingle(Guid submissionId)
```
Comment (JamalAlabdullah) [inline backend/src/Repositories/SubmissionRepository.cs:32]: see comment below 😊 `SubmissionEntity? `

File: backend/src/Repositories/SubmissionRepository.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    public async Task<SubmissionEntity?> Create(SubmissionEntity entity)
+    {
+        _context.Submissions.Add(entity);
+        await _context.SaveChangesAsync();
+        return entity;
+    }
+
+    public async Task<SubmissionEntity?> Delete(Guid submissionId)
```
Comment (JamalAlabdullah) [inline backend/src/Repositories/SubmissionRepository.cs:58]: we also have a waring here , so we can add `?` to line 14

**Review by JamalAlabdullah — CHANGES_REQUESTED** [review]
Good job,😊I

File: backend/src/Repositories/SubmissionRepository.cs (hunk 1 above)
Comment (ErlingHauan) [reply backend/src/Repositories/SubmissionRepository.cs:39]: I did not understand those errors before, but now I do. Thanks for explaining 😄 
Fixed ✅

File: backend/src/Repositories/SubmissionRepository.cs (hunk 2 above)
Comment (ErlingHauan) [reply backend/src/Repositories/SubmissionRepository.cs]: ✅

File: backend/src/Repositories/SubmissionRepository.cs (hunk 3 above)
Comment (ErlingHauan) [reply backend/src/Repositories/SubmissionRepository.cs:58]: ✅

**Comment by ErlingHauan** [comment]
> Good job,😊I

Thanks! 😄

**Pull request merged**